src/main/java/me/jellysquid/mods/lithium/mixin/entity/collision_only_boat_shulker/MixinBoatEntity.java

```diff
@@ -0,0 +1,8 @@
+package me.jellysquid.mods.lithium.mixin.entity.collision_only_boat_shulker;
+
+import me.jellysquid.mods.lithium.common.entity.LithiumEntityCollisions;
+import net.minecraft.entity.vehicle.BoatEntity;
+import org.spongepowered.asm.mixin.Mixin;
+
+@Mixin(BoatEntity.class)
+public class MixinBoatEntity implements LithiumEntityCollisions.CollisionBoxOverridingEntity { }
```

Comment:
Instead of adding marker types to entity classes through Mixins, we should prefer to build a registry in Lithium (a simple HashMap linking EntityType to a special enum would be sufficient). Ultimately we want to avoid Mixins as much as possible and keep it maintainable.